ui/admin-portal/src/app/components/account/change-password/change-password.component.spec.ts

```diff
@@ -0,0 +1,114 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import { ComponentFixture, TestBed, fakeAsync, tick } from '@angular/core/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { ChangePasswordComponent } from './change-password.component';
+import { FormBuilder, ReactiveFormsModule } from '@angular/forms';
+import { NgbActiveModal } from "@ng-bootstrap/ng-bootstrap";
+import { UserService } from '../../../services/user.service';
+import { of } from 'rxjs';
+import { User } from '../../../model/user';
+
+fdescribe('ChangePasswordComponent', () => {
+  let component: ChangePasswordComponent;
+  let fixture: ComponentFixture<ChangePasswordComponent>;
+  let userService: jasmine.SpyObj<UserService>;
+  let activeModal: jasmine.SpyObj<NgbActiveModal>;
+
+  beforeEach(async () => {
+    const userServiceSpy = jasmine.createSpyObj('UserService', ['updatePassword']);
+    const activeModalSpy = jasmine.createSpyObj('NgbActiveModal', ['close', 'dismiss']);
+
+    await TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule,
+        ReactiveFormsModule
+      ],
+      declarations: [
+        ChangePasswordComponent
+      ],
+      providers: [
+        FormBuilder,
+        { provide: UserService, useValue: userServiceSpy },
+        { provide: NgbActiveModal, useValue: activeModalSpy }
+      ]
+    }).compileComponents();
+
+    userService = TestBed.inject(UserService) as jasmine.SpyObj<UserService>;
+    activeModal = TestBed.inject(NgbActiveModal) as jasmine.SpyObj<NgbActiveModal>;
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ChangePasswordComponent);
+    component = fixture.componentInstance;
+    const user: User = {
+      approver: undefined,
+      createdBy: undefined,
+      createdDate: 0,
+      email: "",
+      firstName: "",
+      id: 21221,
+      jobCreator: false,
+      lastLogin: 0,
+      lastName: "",
+      mfaConfigured: false,
+      name: "",
+      partner: undefined,
+      purpose: "",
+      readOnly: false,
+      role: "",
+      sourceCountries: [],
+      status: "",
+      updatedDate: 0,
+      username: "",
+      usingMfa: false
+    };
+    component.user = user;
+    fixture.detectChanges();
+  });
+
+  it('should initialize the form with empty password and passwordConfirmation fields', () => {
+    const formValue = component.form.value;
+    expect(formValue.password).toEqual('');
+    expect(formValue.passwordConfirmation).toEqual('');
+  });
+
+  it('should call updatePassword() function and close modal on successful password update', () => {
+    // Arrange
+    component.form.patchValue({
+      password: 'newPassword',
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We detected credentials hardcoded in your code, which might allow unauthorized users to access your account. We recommend you to store your credentials outside of the code in a configuration file, a database, or a management service for secrets and retrieve from it. For more information, see [CWE-259](https://cwe.mitre.org/data/definitions/259.html) and [CWE-798](https://cwe.mitre.org/data/definitions/798.html).



Suggested remediation:
Do not hardcode password or access key. Instead, try to fetch it from environment to deny unauthorized users to access your account.
```
@@ -92,3 +92,3 @@
     component.form.patchValue({
-      password: &#39;newPassword&#39;,
+      password: process.env.ENV_VARIABLE,
       passwordConfirmation: &#39;newPassword&#39;

```